chroma-agent/chroma_agent/agent_daemon.py

```diff
@@ -207,4 +208,9 @@ def teardown_callback(*args, **kwargs):
         # NB I would rather ensure cleanup by using 'with', but this
         # is python 2.4-compatible code
         context.close()
+
+    # Call any agent daemon startup methods that were registered.
+    for function in agent_daemon_teardown_functions:
```

Comment:
could very well be a comprehension, really , you would want a generator because you would not want/need a list output, yes , we should add cleanup there as well